python/cudf/cudf/core/reshape.py

```diff
@@ -58,6 +83,22 @@ def concat(objs, axis=0, ignore_index=False, sort=None):
 
     # when axis is 1 (column) we can concat with Series and Dataframes
     if axis == 1:
+        # Check if multiindex then check if indexes match. GenericIndex
+        # returns ndarray tuple of bools requiring additional filter.
+        # Then check for duplicate index value.
+        i_objs = iter(objs)
+        first = next(i_objs)
+        if all(isinstance(o.index, cudf.MultiIndex) for o in objs):
+            match_index = all(first.index == rest.index for rest in i_objs)
+        else:
+            match_index = all(
+                all(first.index == rest.index) for rest in i_objs
+            )
```

Comment:
Can we move this logic into the `_align_objs` function? This behavior is a consistent pattern throughout Pandas so we'd want it centralized whereever we're doing this type of alignment.